datanode/datanode.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"log/slog"
+	"math/rand"
+	"net/rpc"
+	"time"
+
+	"github.com/ZhenbangYou/TinyDFS/tiny-dfs/common"
+)
+
+type DataNode struct {
+	nameNodeEndpoint string
+	dataNodeEndpoint string
+}
+
+func (datanode *DataNode) registerWithNameNode() bool {
+	slog.Info("Registering with namenode",
+		"nameNodeEndpoint", datanode.nameNodeEndpoint,
+		"dataNodeEndpoint", datanode.dataNodeEndpoint)
+
+	// retry if failed
+	for {
+		client, err := rpc.DialHTTP("tcp", datanode.nameNodeEndpoint)
+		if err != nil {
+			slog.Error("error dialing namenode", "error", err)
+			// retry after a random interval
+			delay := common.RETRY_MIN_INTERVAL + time.Duration(rand.Int63n(int64(common.RETRY_MAX_INTERVAL-common.RETRY_MIN_INTERVAL)))
```

Comment:
This line is too long. Generally it should not be more than 100 characters (some people will loosen it to 120, but people like John Ousterhout prefer 80).
Two ways to do this:
1. split this line;
2. save part of the expression into another variable, e.g., the random number can be evaluated to another local variable.